contracts/BaseWrapper.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import {RegistryAPI, VaultAPI} from "./BaseStrategy.sol";
+
+abstract contract BaseWrapper {
+    using SafeMath for uint256;
+
+    ERC20 public token;
+
+    // Reduce number of external calls (SLOADs stay the same)
+    VaultAPI[] private _cachedVaults;
+
+    // v2.registry.ychad.eth
+    RegistryAPI constant registry = RegistryAPI(0xE15461B18EE31b7379019Dc523231C57d1Cbc18c);
+
+    constructor(address _token) public {
+        token = ERC20(_token);
+    }
+
+    function bestVault() public virtual view returns (VaultAPI) {
+        return VaultAPI(registry.latestVault(address(token)));
+    }
+
+    function allVaults() public virtual view returns (VaultAPI[] memory) {
+        uint256 cache_length = _cachedVaults.length;
+        uint256 num_deployments = registry.nextDeployment(address(token));
+
+        // Use cached
+        if (cache_length == num_deployments) {
+            return _cachedVaults;
+        }
+
+        VaultAPI[] memory vaults = new VaultAPI[](num_deployments);
+
+        for (uint256 deployment_id = 0; deployment_id < cache_length; deployment_id++) {
+            vaults[deployment_id] = _cachedVaults[deployment_id];
+        }
+
+        for (uint256 deployment_id = cache_length; deployment_id < num_deployments; deployment_id++) {
+            vaults[deployment_id] = VaultAPI(registry.vaults(address(token), deployment_id));
+        }
+
+        return vaults;
+    }
+
+    function _updateVaultCache(VaultAPI[] memory vaults) internal {
+        if (vaults.length > 0) {
+            _cachedVaults = vaults;
+        }
+    }
+
+    function totalBalance(address account) public view returns (uint256 balance) {
+        VaultAPI[] memory vaults = allVaults();
+
+        for (uint256 id = 0; id < vaults.length; id++) {
+            balance = balance.add(vaults[id].balanceOf(account).mul(vaults[id].pricePerShare()).div(10**vaults[id].decimals()));
+        }
+    }
+
+    function totalAssets() public view returns (uint256 assets) {
+        VaultAPI[] memory vaults = allVaults();
+
+        for (uint256 id = 0; id < vaults.length; id++) {
+            assets = assets.add(vaults[id].totalAssets().mul(vaults[id].pricePerShare()).div(10**vaults[id].decimals()));
+        }
+    }
+
+    function _deposit(
+        address depositor,
+        address reciever,
+        uint256 amount,
+        bool pullFunds // If true, funds need to be pulled from `depositor` via `transferFrom`
+    ) internal returns (uint256) {
+        VaultAPI best = bestVault();
+
+        if (pullFunds) token.transferFrom(depositor, address(this), amount);
```

Comment:
weird style is not enforce on 83 same as here

Reply:
Fixed in 2182967